### PR TITLE
stig: use pytestCheckHook framework

### PR DESCRIPTION
**Pull request by ju1m**
###### Motivation for this change
Current package for ˋstigˋ overrides ˋcheckPhaseˋ hence setting ˋdoInstallCheck = falseˋ is ignored.

###### Things done
Use the ˋpytestCheckHookˋ framework.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/p2p/stig/default.nix [hunk 1]
```diff
@@ -38,15 +38,22 @@ python3.pkgs.buildPythonApplication rec {
...
+    pytestCheckHook
   ];
 
-  checkPhase = ''
-    pytest tests
+  dontUseSetuptoolsCheck = true;
+
+  preCheck = ''
+    export LC_ALL=C
+    export HOME=$(mktemp -d)
```
Comment (Mic92) [inline pkgs/applications/networking/p2p/stig/default.nix:50]:
```suggestion
    export HOME=$TMPDIR
```

Comment (ju1m) [reply pkgs/applications/networking/p2p/stig/default.nix:50]: The manual suggests `HOME=$(mkdir -d)`, but both can be found in nixpkgs.
However I introduced it before setting `dontUseSetuptoolsCheck = true;` when I hit the following error:
```
Can't create directory /homeless-shelter/.local/share/stig/histories: Permission denied
```
But with `dontUseSetuptoolsCheck = true;` it doesn't fail, so let's remove it I guess.

File: pkgs/applications/networking/p2p/stig/default.nix [hunk 2]
```diff
@@ -38,17 +38,22 @@ python3.pkgs.buildPythonApplication rec {
...
-      --deselect=tests/client_test/ttypes_test.py::TestTimestamp::test_string__month_day_hour_minute_second
+  dontUseSetuptoolsCheck = true;
+
+  preCheck = ''
+    export LC_ALL=C
   '';
 
+  pytestFlagsArray = [
+    "tests"
+    "--deselect=tests/client_test/ttypes_test.py::TestTimestamp::test_string__month_day_hour_minute_second"
```
Comment (doronbehar) [inline pkgs/applications/networking/p2p/stig/default.nix:54]: The comment that was there from before.

```suggestion
    # test_string__month_day_hour_minute_second fails on darwin
    "--deselect=tests/client_test/ttypes_test.py::TestTimestamp::test_string__month_day_hour_minute_second"
```

**Review by doronbehar — CHANGES_REQUESTED** [review]
Besides that, builds and runs fine for me.

**Comment by ju1m** [comment]
Thanks @doronbehar.
Also, FYI, I've asked upstream for a new release (last one dates back to 2018). https://github.com/rndusr/stig/issues/133

**Review by doronbehar — APPROVED** [review]

**Pull request merged**